### ISSUE TITLE
Docker v1.8.0-rc1cannot be build on ARM 32bit

### ISSUE THREAD

**Issue by DieterReuter** [docker/docker#15053]
Trying to build Docker 1.8.0-rc1 on a Scaleway server (ARM 32bit, armv7l) failed:
```
# Try this instead: make all
#
---> Making bundle: dynbinary (in bundles/1.8.0-rc1/dynbinary)
# github.com/opencontainers/runc/libcontainer/seccomp
vendor/src/github.com/opencontainers/runc/libcontainer/seccomp/filter.go:93: undefined: jumpEqualTo
vendor/src/github.com/opencontainers/runc/libcontainer/seccomp/filter.go:95: undefined: jumpNotEqualTo
vendor/src/github.com/opencontainers/runc/libcontainer/seccomp/filter.go:97: undefined: jumpGreaterThan
vendor/src/github.com/opencontainers/runc/libcontainer/seccomp/filter.go:99: undefined: jumpLessThan
vendor/src/github.com/opencontainers/runc/libcontainer/seccomp/filter.go:101: undefined: jumpMaskEqualTo
```
The `seccomp` package has currently a `jump_amd64.go` file for linux/amd64 only!

**Comment by thaJeztah** [comment]
Looks like this problem is in vendored files, and needs changes in the runc repository, could you report there as well?

**Comment by jefby** [comment]
arm64 alse compile error , ARMv8 cortex-a57

**Comment by paralin** [comment]
There's a simple patch for this. See the buildroot recepie I wrote found in https://github.com/paralin/buildroot

**Comment by thaJeztah** [comment]
@paralin are those patches suitable for inclusion in Docker by default? IIRC, plans are to eventually make Docker build on ARMv7/v8 without the need to apply patches, if there's a clean way to integrate those, we welcome PR's (but it may make sense to discuss in the #docker-dev IRC channel before that).

**Comment by paralin** [comment]
It's a very simple set of patches, I'll submit PRs.

**Comment by domenkozar** [comment]
Upstream has a different source: https://github.com/opencontainers/runc/tree/master/libcontainer/seccomp

**Comment by domenkozar** [comment]
I've fixed it in NixOS with https://github.com/NixOS/nixpkgs/commit/e65fce3af64ea0b2058a69d72bc74a0baa756bf9

**Comment by fedya** [comment]
@domenkozar thanks for your commit, really helped me to fix similar issue on armv7hl port.

**Comment by DieterReuter** [comment]
I'll close this issues as it is already fixed in Docker 1.9.0, for 1.8.x it's still present.
